### PR TITLE
upgrade to 0.43

### PR DESCRIPTION
**Pull request by zmanian**
WIP pull request to upgrade to cosmos sdk 0.43-beta1.

Go tests still don't pass because of changes to the bank api.

Working on it.

### CODE REVIEW

File: module/x/gravity/keeper/pool_test.go [hunk 1]
```diff
@@ -25,7 +25,8 @@ func TestAddToOutgoingPool(t *testing.T) {
 
 	// set senders balance
 	input.AccountKeeper.NewAccountWithAddress(ctx, mySender)
-	err = input.BankKeeper.SetBalances(ctx, mySender, allVouchers)
+
```
Comment (albertchon) [inline module/x/gravity/keeper/pool_test.go:28]: Maybe you forgot the mint call here

**Comment by zmanian** [comment]
@levicook can we also try merging this into your branch and look for regressions?

File: module/x/gravity/keeper/test_common.go [hunk 2]
```diff
@@ -362,21 +376,22 @@ func CreateTestEnv(t *testing.T) TestInput {
 
 	// set genesis items required for distribution
 	distKeeper.SetFeePool(ctx, distrtypes.InitialFeePool())
+	// accountKeeper.SetModuleAccount(ctx, stakingKeeper.GetNotBondedPool(ctx))
+	// accountKeeper.SetModuleAccount(ctx, stakingKeeper.GetBondedPool(ctx))
```
Comment (mvid) [inline module/x/gravity/keeper/test_common.go:380]: Do we want to keep commented code?

File: module/x/gravity/types/msgs.go [hunk 3]
```diff
@@ -39,6 +39,7 @@ func (msg *MsgDelegateKeys) Type() string { return "delegate_keys" }
 
 // ValidateBasic performs stateless checks
 func (msg *MsgDelegateKeys) ValidateBasic() (err error) {
+	fmt.Println(msg)
```
Comment (mvid) [inline module/x/gravity/types/msgs.go:42]: This was probably left in after debugging

File: module/x/gravity/types/msgs_test.go [hunk 4]
```diff
@@ -59,8 +66,9 @@ func TestValidateMsgDelegateKeys(t *testing.T) {
 			expErr:        true,
 		},
 	}
-	for msg, spec := range specs {
-		t.Run(msg, func(t *testing.T) {
+	for test_msg, spec := range specs {
+		t.Run(test_msg, func(t *testing.T) {
+			fmt.Println(test_msg)
```
Comment (mvid) [inline module/x/gravity/types/msgs_test.go:71]: If this is worth keeping in test output, we should probably use t.Log

File: module/x/gravity/types/msgs_test.go [hunk 5]
```diff
@@ -73,3 +81,20 @@ func TestValidateMsgDelegateKeys(t *testing.T) {
 	}
 
 }
+
+func VerifyAddressFormat(bz []byte) error {
```
Comment (mvid) [inline module/x/gravity/types/msgs_test.go:85]: Does this function need to be duplicated from module/app/app.go?

**Comment by levicook** [comment]
> @levicook can we also try merging this into your branch and look for regressions?

I don’t have anything open / not on main as of Friday morning. Depending on when you created this branch, you might just need to rebase to pickup everything that has landed.

File: module/cmd/gravity/cmd/gentx.go [hunk 6]
```diff
@@ -86,10 +86,10 @@ $ %s gentx my-key-name 1000000stake 0x033030FEeBd93E3178487c35A9c8cA80874353C9 c
 			}
 
 			// read --pubkey, if empty take it from priv_validator.json
-			if valPubKeyString, _ := cmd.Flags().GetString(cli.FlagPubKey); valPubKeyString != "" {
-				valPubKey, err = sdk.GetPubKeyFromBech32(sdk.Bech32PubKeyTypeConsPub, valPubKeyString)
+			if val, _ := cmd.Flags().GetString(cli.FlagPubKey); val != "" {
+				err = clientCtx.JSONCodec.UnmarshalJSON([]byte(val), valPubKey)
```
Comment (fedekunze) [inline module/cmd/gravity/cmd/gentx.go:90]: curious about why this logic changed(?)

File: module/x/gravity/cosmos_originated_test.go [hunk 7]
```diff
@@ -38,12 +38,14 @@ type testingVars struct {
 	t                  *testing.T
 }
 
+const AddrLen = 20
```
Comment (fedekunze) [inline module/x/gravity/cosmos_originated_test.go:41]: we can use this const from either the SDK or geth types

File: module/x/gravity/keeper/keeper_test.go [hunk 8]
```diff
@@ -12,6 +12,8 @@ import (
 	"github.com/cosmos/gravity-bridge/module/x/gravity/types"
 )
 
+const Addrlen = 20
```
Comment (fedekunze) [inline module/x/gravity/keeper/keeper_test.go:15]: ditto

File: module/x/gravity/types/msgs_test.go [hunk 9]
```diff
@@ -2,17 +2,24 @@ package types
...
+	"fmt"
 	"testing"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/address"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 	"github.com/stretchr/testify/assert"
 )
 
+const AddrLen = 20
```
Comment (fedekunze) [inline module/x/gravity/types/msgs_test.go:14]: ditto

File: module/x/gravity/types/msgs_test.go (hunk 5 above)
Comment (fedekunze) [reply module/x/gravity/types/msgs_test.go:85]: I wonder if the SDK exposes this function at all

**Review by fedekunze — APPROVED** [review]
LGTM/utACK

File: module/x/gravity/types/msgs_test.go (hunk 5 above)
Comment (zmanian) [reply module/x/gravity/types/msgs_test.go:85]: So here is the deal.

The SDK's default verifier no longer errors on short addresses. This test's behavior expects errors on short addresses.

SDK address verifiers are global functions so the proper place to put the verifier is app.go and not in the module.

it's not entirely clear what the best way to deal with this change in the SDK behavior is.

File: module/x/gravity/cosmos_originated_test.go (hunk 7 above)
Comment (zmanian) [reply module/x/gravity/cosmos_originated_test.go:41]: this const isn't in the SDK anymore.

**Pull request closed without merging**